search.go

```diff
@@ -266,18 +267,40 @@ func (s *Search) FullSearch(q string, fuzzy bool) ([]*SearchHit, int, time.Durat
 	return hits, int(res.Total), res.Took, s.isStale, nil
 }
 
+type Nodes []*Node
+
+// String matches https://github.com/sahilm/fuzzy Source interface
+func (n Nodes) String(i int) string {
+	return n[i].Title()
+}
+
+// Len matches https://github.com/sahilm/fuzzy Source interface
+func (n Nodes) Len() int {
+	return len(n)
+}
```

Comment:
To allow searching for more than just the title, I'd like to make this more flexible.  The new type should map node Ref-IDs (aka URL) to different parts of the node we'd like to fuzzy match against.

In pseudo-Go:

```
type FuzzySearchSource {
  // Implement Add(nodeURL, someStringToMatch)
  // implement String
  // implement Len
}

source := FuzzySearchSource()
for _, n := range nodes {
  source.Add(n.URL(), n.Title())
  for _, tag := range n.Tags() {
    source.Add(n.URL(), tag)
  }
}
```

Reply:
This will probably return multiple matches per node, so that after `Find()` the matches need to iterated and matches be filtered out:

```
Components/List: "foo from the title"
abc/def: "foobar"
Components/List: "foo as a tag" // this is filtered out, as the node appeared earlier
```

Reply:
I'm still pretty torn about where the line between search and filter is. Considering the response format, I'd like to see that line pushed all the way to "filter only works on titles."

If we add arbitrary fields, but then don't intend give the API consumer that feedback, we should _only_ use those fields in search.